src/include/crowbar/complex.rb

```diff
@@ -695,12 +617,195 @@ def StoreRepoURL(key, event)
     def HandleRepoURL(key, event)
       # store the value on exiting
       if event["ID"] == :next || event["EventReason"] == "ValueChanged"
+        @current_repo = UI.QueryWidget(Id("repos_table"), :Value) || ""
         StoreRepoURL(key, event)
         InitReposTable("repos_table")
       end
       nil
     end
 
+    def InitLocationURL(id)
+      UI.ChangeWidget(Id(id), :Value, @remote_server_url)
+      nil
```

Comment:
So I was curious, and Jiri posted a backtrace of a crash that he got after removing nils:

```
2015-07-22 17:09:58 <1> linux-bfa7.site(28160) [Ruby] modules/Progress.rb:512 No progress icons defined
2015-07-22 17:10:26 <2> linux-bfa7.site(28160) [Ruby] (eval):2 Cannot convert TrueClass from 'any' to 'symbol'
2015-07-22 17:10:26 <2> linux-bfa7.site(28160) [Ruby] (eval):2 ------------- Backtrace begin -------------
2015-07-22 17:10:26 <2> linux-bfa7.site(28160) [Ruby] (eval):2 (eval):2:in `to_symbol'
2015-07-22 17:10:26 <2> linux-bfa7.site(28160) [Ruby] (eval):2 /usr/share/YaST2/modules/CWM.rb:852:in `Run'
2015-07-22 17:10:26 <2> linux-bfa7.site(28160) [Ruby] (eval):2 /usr/share/YaST2/modules/CWM.rb:938:in `ShowAndRun'
2015-07-22 17:10:26 <2> linux-bfa7.site(28160) [Ruby] (eval):2 /home/jsuchome/yast-crowbar/src/include/crowbar/complex.rb:1727:in `OverviewDialog'
```

The answer is: CWM does expect the handlers to return a Symbol or a `nil`. So those methods did not have `void` as the return type in the first place, and the `nil` should not be removed.